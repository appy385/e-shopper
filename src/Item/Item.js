import React from 'react';
import PropTypes from 'prop-types';
import './Item.scss';

const Item = ({ product }) => {
  return (
    <tr className="item">
      <td>{product.productName}</td>
      <td>{product.price}</td>
      <td>{product.count}</td>
      <td data-testid="item-amount">{product.count * product.price}</td>
    </tr>
  );
}

Item.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number.isRequired,
    src: PropTypes.string,
    seller: PropTypes.string,
    productName: PropTypes.string.isRequired,
    quantity: PropTypes.string,
    price: PropTypes.number.isRequired,
    count: PropTypes.number.isRequired,
    category: PropTypes.string.isRequired,
  }).isRequired,

};

export default Item;
