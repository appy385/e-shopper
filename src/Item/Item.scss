.item>td {
    border-top: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    padding: 15px;
    text-align: left;
  
}