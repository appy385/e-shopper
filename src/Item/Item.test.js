import React from 'react';
import { render, screen } from '@testing-library/react';
import Item from './Item';

describe(Item.name, ()=>{
    let mockProduct;
    beforeEach(() => {
        mockProduct = {
          id: 1,
          src: 'https://images.pexels.com/photos/61127/pexels-photo-61127.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500',
          seller: 'Fresho',
          productName: 'Banana-Robusta',
          quantity: '1kg',
          price: 40,
          count: 2,
          category: 'Fruits & Vegatables',
        };
     });
    afterEach(() => {
        jest.clearAllMocks();
    });
    test('should match snapshot for Item component', () => {
        const { container } = render(<table><tbody><Item
          product={mockProduct}
        /></tbody></table>);
        expect(container).toMatchSnapshot();
    });

    test('should display the product details', () => {
        render(<table><tbody><Item
          product={mockProduct}
        /></tbody></table>);
        const amount = mockProduct.price *mockProduct.count
        screen.getByText(mockProduct.productName);
        screen.getByText(mockProduct.price);
        screen.getByText(mockProduct.count);
        const ele = screen.getByTestId("item-amount");
        // expect(ele.).toEqual();
      });
    
})