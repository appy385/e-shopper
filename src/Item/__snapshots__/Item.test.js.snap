// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should match snapshot for Item component 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="item"
      >
        <td>
          Banana-Robusta
        </td>
        <td>
          40
        </td>
        <td>
          2
        </td>
        <td
          data-testid="item-amount"
        >
          80
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
